system/src/system_power_manager.cpp

```diff
@@ -927,7 +936,15 @@ void PowerManager::deinit() {
     }
   }
 
-  detachInterrupt(LOW_BAT_UC);
+  hal_pin_t pmicIntPin = LOW_BAT_UC;
+#if PLATFORM_ID == PLATFORM_MSOM
+  uint32_t revision;
+  hal_get_device_hw_version(&revision, nullptr);
+  if (revision == 1) {
```

Comment:
Should we define this magic number for MSoM EVT hardware somewhere in DVOS?

Reply:
Okay. I misread the question. So I think we should define all the hardware revisions in a common header file for all existing platforms. It would be better to leave a comment for each macro to elabrate the hardware changes per revision, so that we can easily understand what has been changed over time.

Reply:
> Okay. I misread the question. So I think we should define all the hardware revisions in a common header file for all existing platforms. It would be better to leave a comment for each macro to elabrate the hardware changes per revision, so that we can easily understand what has been changed over time.

I agree. This is the second time we need to be aware of a specific platform and hardware revision. The first time was for the boron ADC change [here](https://github.com/particle-iot/device-os/blob/develop/hal/src/nRF52840/adc_hal.cpp#L35)

Where would be a good spot to add this header? `hal/inc/hardware_revisions.h` or something?